pkg/api/norman/store/cluster/cluster_store.go

```diff
@@ -306,14 +308,32 @@ func loadDataFromTemplate(clusterTemplateRevision *v3.ClusterTemplateRevision, c
 			return nil, httperror.WrapAPIError(err, httperror.ServerError, "Error processing clusterTemplate answers")
 		}
 		keyParts := strings.Split(question.Variable, ".")
-		values.PutValue(dataFromTemplate, val, keyParts...)
+		if strings.HasPrefix(question.Variable, registryConfigPath) {
+			key := keyParts[len(keyParts)-1]
+			index, err := strconv.Atoi(string(question.Variable[len(registryConfigPath)+1]))
```

Comment:
This is pretty hacky and requires at least a comment - that we get a digit from `rancherKubernetesEngineConfig.privateRegistries[0].password` this way. It's not trivial by looking at it.
Overall this does look a bit hacky and I was hoping there is a cleaner way to extract index that works not only when there is a single digit.

Reply:
A new logic is added to handle the index correctly.